src/misc-utils.ts

```diff
@@ -1,6 +1,26 @@
-import { isErrorWithCode, wrapError } from '@metamask/utils/node';
-import type fs from 'fs';
-import * as fsPromises from 'fs/promises';
+import { isErrorWithCode, isObject } from '@metamask/utils/node';
+import type { Json } from '@metamask/utils/node';
+import fs from 'fs';
+import path from 'path';
+import { ErrorWithCause } from 'pony-cause';
+import { StructError, assert } from 'superstruct';
+import type { Struct } from 'superstruct';
+import type { ObjectSchema } from 'superstruct/dist/utils';
+
+/**
+ * Represents a directory entry. Like fs.Dirent, but includes a `path` property
```

Comment:
Nit: It seems like the `path` property is unchanged from the `Dirent` object, but rather there are _two additional_ properties for the "full" (maybe should be called "absolute"?) and relative path

Reply:
Ah, true, thanks. JSDoc updated here: [685621e](https://github.com/MetaMask/module-lint/pull/39/commits/685621e04ee299e7055b2d5f24288a22c71036d9)